test/k8sT/Services.go

```diff
@@ -1548,6 +1549,155 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 		})
 	})
 
+	SkipContextIf(func() bool {
+		// The graceful termination feature depends on enabling an alpha feature
+		// EndpointSliceTerminatingCondition in Kubernetes.
+		return helpers.RunsOnGKE() || helpers.RunsOnEKS()
+	}, "Checks graceful termination of service endpoints", func() {
+		const (
+			clientPodLabel = "app=graceful-term-client"
+			serverPodLabel = "app=graceful-term-server"
+			testPodLabel   = "zgroup=testDSClient"
+		)
+		var (
+			gracefulTermYAML string
+			clientPod        string
+			serverPod        string
+			wg               sync.WaitGroup
+		)
+
+		waitForServer := func() {
```

Comment:
Could you move the helper functions to `service_helpers.go`, so that the file size can be reduced for the sake of readability? Also, you could make it more generic so that it can be reused.